examples/spouse/app.ddlog

```diff
@@ -0,0 +1,243 @@
+  person_mention(p2_id, _, doc_id, sent_index, p2_begin_index, p2_end_index),
+  sentences(doc_id, sent_index, _, tokens, lemmas, pos_tags, ner_tags, _, dep_types, dep_tokens
+).
+
+
+## Distant Supervision ########################################################
+@extraction
+spouse_label(
+    @key
+    @references(relation="has_spouse", column="p1_id", alias="has_spouse")
+    p1_id text,
+    @key
+    @references(relation="has_spouse", column="p2_id", alias="has_spouse")
+    p2_id text,
+    @navigable
+    label int,
+    @navigable
+    rule_id text
+).
+
+# make sure all pairs in spouse_candidate are considered as unsupervised examples
+spouse_label(p1, p2) = NULL :- spouse_candidate(p1, _, p2, _).
+
+# distant supervision using data from DBpedia
+spouse_label(p1,p2, 1, "from_dbpedia") :-
+  spouse_candidate(p1, p1_name, p2, p2_name), spouses_dbpedia(n1, n2),
+  [ lower(n1) = lower(p1_name), lower(n2) = lower(p2_name) ;
+    lower(n2) = lower(p1_name), lower(n1) = lower(p2_name) ].
+
+# supervision by heuristic rules in a UDF
+function supervise over (
+        p1_id text, p1_begin int, p1_end int,
+        p2_id text, p2_begin int, p2_end int,
+        doc_id         text,
+        sentence_index int,
+        sentence_text  text,
+        tokens         text[],
+        lemmas         text[],
+        pos_tags       text[],
+        ner_tags       text[],
+        dep_types      text[],
+        dep_tokens    int[]
+    ) returns (
+        p1_id text, p2_id text, label int, rule_id text
+    )
+    implementation "udf/supervise_spouse.py" handles tsv lines.
+
+spouse_label += supervise(
+    p1_id, p1_begin, p1_end,
+    p2_id, p2_begin, p2_end,
+    doc_id, sentence_index, sentence_text,
+    tokens, lemmas, pos_tags, ner_tags, dep_types, dep_token_indexes
+) :- spouse_candidate(p1_id, _, p2_id, _),
+    person_mention(p1_id, p1_text, doc_id, sentence_index, p1_begin, p1_end),
+    person_mention(p2_id, p2_text,      _,              _, p2_begin, p2_end),
+    sentences(
+        doc_id, sentence_index, sentence_text,
+        tokens, lemmas, pos_tags, ner_tags, _, dep_types, dep_token_indexes
+    ).
+
+
+# resolve multiple labels by majority vote (summing the labels in {-1,0,1})
+spouse_label_resolved(p1_id, p2_id, SUM(vote)) :- spouse_label(p1_id, p2_id, vote, rule_id).
+
+# assign the resolved labels for the spouse relation
+has_spouse(p1_id, p2_id) = if l > 0 then TRUE
+                      else if l < 0 then FALSE
+                      else NULL end :- spouse_label_resolved(p1_id, p2_id, l).
+
+###############################################################################
+
+## Inference Rules ############################################################
+
+# Features
+@weight(f)
+has_spouse(p1_id, p2_id) :-
+  spouse_candidate(p1_id, _, p2_id, _),
+  spouse_feature(p1_id, p2_id, f).
+
+# Inference rule: Symmetry
+@weight(3.0)
+has_spouse(p1_id, p2_id) => has_spouse(p2_id, p1_id) :-
```

Comment:
I agree.  I think the only reason implication appeared here is because the other example we looked at was written before we added the Equals-factor syntax.